packages/react/src/hooks/useStorageURL.ts

```diff
@@ -14,31 +14,35 @@ export interface UseStorageURLResult {
  */
 export const useStorageURL = (
   key: string,
-  options?: S3ProviderGetConfig
+  options?: S3ProviderGetConfig,
+  fallbackURL?: string
 ): UseStorageURLResult & { fetch: () => () => void } => {
   const [result, setResult] = React.useState<UseStorageURLResult>({
     isLoading: true,
   });
 
   // Used to prevent an infinite loop on useEffect, because `options`
   // will have a different reference on every render
-  const serializedOptions = JSON.stringify(options);
+  const optionsRef = React.useRef(options);
```

Comment:
What if `options` changes?

Reply:
It is programmatically possible. In reality, is there a use case that users would update the configs dynamically with the same object? 😅

Reply:
> In reality, is there a use case that users would update the configs dynamically with the same object?

If the user passes an updated `options` then the output is going to be incorrect since the new `options` will just be ignored. Also this is a breaking change from the current implementation

Reply:
We never use the serialized one and it causes infinite loop at the beginning 😅  <- https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/hooks/useStorageURL.ts#L25-L41

I can make the change to serialize the object but still think the use case is rare. It feels strange that you would update the options programmatically in an app without changing the key to another object. Like in [`useDataStore` hook](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/hooks/useDataStore.tsx#L50), we set the dependency array to `[]` though some other configs may change

Reply:
Agree to make it be part of the dependency. But if we do not serialize it, how could you avoid infinite loop?

Reply:
The consuming component can just memoize `options` before passing it to `useStorageUrl`

Reply:
That's true but `useStorageURL` could be used somewhere else as a hook. Think it should handle the case internally instead of leaving the issue to its consumers?

Reply:
Then we can do whatever we want